yt_dlp/extractor/mxplayer.py

```diff
@@ -28,12 +28,41 @@ class MxplayerIE(InfoExtractor):
             'skip_download': True,
             'format': 'bestvideo'
         }
-    }
+    }, {
+        'url': 'https://www.mxplayer.in/show/watch-shaitaan/season-1/the-infamous-taxi-gang-of-meerut-online-45055d5bcff169ad48f2ad7552a83d6c',
+        'info_dict': {
+            'id': '45055d5bcff169ad48f2ad7552a83d6c',
+            'ext': 'm3u8',
+            'title': 'The infamous taxi gang of Meerut',
+            'description': 'md5:033a0a7e3fd147be4fb7e07a01a3dc28',
+            'season': 'Season 1',
+            'series': 'Shaitaan'
+        },
+        'params': {
+            'skip_download': True,
+        }
+
+    }]
 
     def _get_best_stream_url(self, stream):
         best_stream = list(filter(None, [v for k, v in stream.items()]))
         return best_stream.pop(0) if len(best_stream) else None
 
+    def _get_stream_urls_shows(self, video_dict):
+        stream_dict = video_dict.get('stream', {'provider': {}})
+        stream_provider = stream_dict.get('provider')
+        stream_provider_dict = stream_dict.get(stream_provider)
+        if not stream_dict[stream_provider]:
+            message = 'No stream provider found'
+            raise ExtractorError('%s said: %s' % (self.IE_NAME, message), expected=True)
+
+        streams = []
+        if stream_provider_dict['dashUrl']:
+            streams.append(('dash', stream_provider_dict['dashUrl']))
+        if stream_provider_dict['hlsUrl']:
+            streams.append(('hls', stream_provider_dict['hlsUrl']))
+        return streams
+
```

Comment:
A lot of code reused from `_get_stream_urls`. Generalize into a single function

Reply:
only the lines 52,53 and 55,56,57 are same.

Movies and shows serve the url differently.
Movies serve them as a dict with 'hls' or 'dash' name whereas with shows, it's just a key value pair.

Reply:
So only L60-63 are different. You can pass the `type` into the function and easily put this in a if condition

Reply:
got it